app/models/release.rb

```diff
@@ -177,7 +178,26 @@ class Release < ApplicationRecord
 
   delegate :versioning_strategy, :patch_version_bump_only, to: :train
   delegate :app, :vcs_provider, :release_platforms, :notify!, :continuous_backmerge?, :approvals_enabled?, :copy_approvals?, to: :train
-  delegate :platform, :organization, to: :app
+  delegate :platform, :organization, :notification_provider, to: :app
+
+  def update_notification_channel!
+    notification_channel = notification_provider.create_channel!(channel_name)
+    train.update(notification_channel:)
+    train.notification_settings.update(notification_channels: [notification_channel])
+  end
+
+  def channel_name
```

Comment:
I think we should rename this method or inline it in the above function. `channel_name` is confusing and doesn't make much sense in the context of a release.